src/prefect/cli/deployment.py

```diff
@@ -933,7 +934,20 @@ async def delete(
         $ prefect deployment delete --id dfd3e220-a130-4149-9af6-8d487e02fea6
     """
     async with get_client() as client:
-        if name is None and deployment_id is not None:
+        if _all:
+            deployments = await client.read_deployments()
+            if len(deployments) == 0:
+                exit_with_error("No deployments found.")
+            if is_interactive() and not typer.confirm(
+                f"Are you sure you want to delete all {len(deployments)} deployments?",
+                default=False,
+            ):
+                exit_with_error("Deletion aborted.")
+            for deployment in deployments:
+                await client.delete_deployment(deployment.id)
+            plural = "" if len(deployments) == 1 else "s"
+            exit_with_success(f"Deleted {len(deployments)} deployment{plural}.")
+        elif name is None and deployment_id is not None:
```

Comment:
minor readability point but given that all branches of the previous `if _all:` exit, i think this reads simpler if this becomes its own `if name is None and deployment_id is not None:` instead of an `elif`.